src/hooks/youtube/video/useSingleYoutubeVideo.js

```diff
@@ -0,0 +1,44 @@
+import { useReducer } from 'react';
+import { youtubeVideoReducer } from './youtubeVideoReducer';
+import {
+  SINGLE_VIDEO_FAIL,
+  SINGLE_VIDEO_REQUEST,
+  SINGLE_VIDEO_SUCCESS,
+} from './actionTypes';
+
+const initialState = {
+  videoId: null,
+  isLoading: false,
+  relatedVideos: [],
+  nextPageToken: null,
+};
+
+export function useSingleYoutubeVideo({ videoId }) {
+  const [state, dispatch] = useReducer(youtubeVideoReducer, initialState);
```

Comment:
very nice i liked that you took out the complexity of the state management to a reducer, 👍